gotemplate.go

```diff
@@ -27,8 +27,8 @@ type GoTemplatePrinter struct {
 }
 
 // NewGoTemplatePrinter returns a printer for outputting values in JSON format.
-func NewGoTemplatePrinter(tmpl string) (ValuePrinter, error) {
-	t, err := template.New("template").Parse(tmpl)
+func NewGoTemplatePrinter(tmpl string, funcMap template.FuncMap) (ValuePrinter, error) {
```

Comment:
Since this module is 1.x (and not 0.x) we should not break the existing API while keeping the semver major but instead have, say, `NewGoTemplatePrinterWithFuncs` (or `...FuncMap`, ...). I don't think that we need a v2, as there always will be no-funcmap usages.

Reply:
@thediveo Makes sense. I didn't realize `NewGoTemplatePrinter` was public API. I assumed users will only be able to use via `klo.Specs`.

I have updated the PR by adding a separate `NewGoTemplatePrinterWithFuncs` function instead of changing existing `NewGoTemplatePrinter`.